host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1089,11 +1090,21 @@ static FILE *open_filename(RASPIVID_STATE *pState, char *filename)
    {
       // Create a new filename string
 
-      // If %d or %u, assume a segment numnber, otherwise use the formatter as
-      // input to strnftime
-      if (strstr(filename,"%u") != NULL || strstr(filename,"%d") != NULL)
-      {
-          asprintf(&tempname, filename, pState->segmentNumber);
+	  //If %d/%u or any valid combination e.g. %04d is specified, assume segment number.
+	  bool bSegmentNumber = false;
+	  const char* pPercent = strstr(filename, "%");
```

Comment:
```strchr(filename, '%');``` would be ever so slightly more efficient as it's only looking to match a single char rather than the string.

Reply:
Nice! I will make this change :)